opencti-platform/opencti-graphql/src/domain/user.js

```diff
@@ -77,6 +78,25 @@ const AUTH_BASIC = 'BasicAuth';
 export const TAXIIAPI = 'TAXIIAPI';
 const PLATFORM_ORGANIZATION = 'settings_platform_organization';
 
+export const userConfidenceSchema = {
+  type: 'object',
+  properties: {
+    max_confidence: { type: 'number' },
+    overrides: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          entity_type: { type: 'string' },
+          max_confidence: { type: 'number' }
+        },
+        required: ['entity_type', 'max_confidence']
+      }
+    }
+  },
+  required: ['max_confidence', 'overrides']
```

Comment:
how will we update this object with the api ? will we need to replace the whole object if we want to update only max_confidence ? I just tried to do a fieldPatch on user_confidence_level and it won't work if I want to only update max_confidence, I need to provide the whole object.

I tried this first (fails with a schema validation error) : 
```
fieldPatch(
      input: {
        key: "user_confidence_level"
        object_path: "max_confidence"
        value: [80]
      }
    )
``` 

Then I tried this, which fails because overrides is not present (and it's wrong because I don't want to replace the whole object) : 
``` 
fieldPatch(
      input: {
        key: "user_confidence_level"
        value: [{
          max_confidence: 80,
        }]
      }
    )
```

Reply:
field patch is actually not implemented for non-multiple objects in schema, it was only done in the context of TAI's weight and height fields.

A large refactoring is ongoing in the schema validation/mapping area, so we will wait for this and sync afterwards.

Meanwhile, let's just update the full field at once, not partially.

I'll remove the UnsupportedOperation error for non-multiple object update in middleware.js, as it works fine now the that the schema-validator properly checks them.